addon-test-support/scroll.js

```diff
@@ -0,0 +1,47 @@
+import getElementWithAssert from './-private/get-element-with-assert';
+import wait from 'ember-test-helpers/wait';
+
+
+/*
+  Triggers a paint (and therefore flushes any queued up scroll events).
+
+  @method triggerFlushWithPromise
+  @return {RSVP.Promise}
+  @private
+*/
+function triggerFlushWithPromise(){
+  return wait().then(() => {
```

Comment:
In all honesty, I understand it enough to rationalize it in my head, not defend it -- so that's my bad.
@pzuraq can you chime in here? That way we add the explanation to the documentation and we all can understand why